torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -14,6 +14,81 @@ struct NoneValue : SugaredValue {
   }
 };
 
+std::shared_ptr<SugaredValue> MethodValue::call(
+    const SourceRange& loc,
+    Function& caller,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    size_t n_binders) {
+
+  Graph& graph = *caller.graph();
+  auto& callee = *method_;
+  try {
+    callee.ensure_defined();
+  } catch (std::exception&) {
+    throw ErrorReport(loc)
+        << " method '" << callee.name()
+        << "' is called recursively involving this call site. Recursive calls are not supported";
+  }
+
+  auto callee_graph = callee.graph();
+  auto& schema = callee.getSchema();
+
+  std::vector<NamedValue> inputsWithSelf(inputs.begin(), inputs.end());
+  auto self_val = self_.value(*caller.graph());
```

Comment:
Self values are always present. This was refactored a few weeks ago.

Reply:
Also, this does not replace FunctionValue::call which the root place where function calls get inserted. MethodValue should be unmodified in this patch. This code should be in FunctionValue::call

Reply:
This also seems like the wrong place to put this code. We have a method on `Function` with the name `emit_call_to`. I feel like this implementation should be there. This also explains why this code seems so long: it is copying all of the logic that already exists in that function to a new place without deleting it.

Reply:
okay... let me see if I can move stuff around